strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/indexing/remote/RepositoryProxyIndexFounder.java

```diff
@@ -0,0 +1,127 @@
+package org.carlspring.strongbox.storage.indexing.remote;
+
+import org.carlspring.strongbox.providers.io.RepositoryFiles;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.io.RepositoryPathResolver;
+import org.carlspring.strongbox.service.ProxyRepositoryConnectionPoolConfigurationService;
+import org.carlspring.strongbox.storage.indexing.*;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexDirectoryPathResolver.RepositoryIndexDirectoryPathResolverQualifier;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexFounder.RepositoryIndexFounderQualifier;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexingContextFactory.RepositoryIndexingContextFactoryQualifier;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryTypeEnum;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Objects;
+
+import org.apache.maven.index.incremental.DefaultIncrementalHandler;
+import org.apache.maven.index.updater.DefaultIndexUpdater;
+import org.apache.maven.index.updater.IndexUpdateRequest;
+import org.apache.maven.index.updater.IndexUpdateResult;
+import org.apache.maven.index.updater.IndexUpdater;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author carlspring
+ * @author Przemyslaw Fusik
+ */
+@Component
+@RepositoryIndexFounderQualifier(RepositoryTypeEnum.PROXY)
+public class RepositoryProxyIndexFounder
```

Comment:
Cosmetic: Please, rename this accordingly. :)

Reply:
done

Reply:
Thanks! :)